### PR TITLE
Pipeline quota fixes

### PR DESCRIPTION
**Pull request by gitlawr**
Related to:
https://github.com/rancher/rancher/issues/16538
https://github.com/rancher/rancher/issues/18716
https://github.com/rancher/rancher/issues/18727

This PR address following issues:

1. Pipeline resource creation is blocked by default zero quotas. Waiting for the namespace condition/status cannot guarantee that because there is still a gap between condition is set and the resource quota is updated. 
Solution: Do retry for resource creation to avoid that.
2. The default quota limit breaks previous pipelines that exceed the limit. There is no reasonable default quota for buildimage and runscript steps wich run user-defined scripts.
Solution: Remove the default resources for those two types.
3. Invalid resources configs cause panic
Solution: Avoid using the MustParse function

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go [hunk 1]
```diff
@@ -65,7 +65,15 @@ func (l *Lifecycle) deploy(projectName string) error {
...
+	for i := 0; i < 3; i++ {
+		// If project resource quota is enabled, It won't succeed until the resource quota in namespace is synced.
+		// Do retries for this case.
+		if _, err = l.secrets.Create(secret); err == nil || apierrors.IsAlreadyExists(err) {
+			break
+		}
+		time.Sleep(2 * time.Second)
+	}
+	if err != nil && !apierrors.IsAlreadyExists(err) {
 		return errors.Wrapf(err, "Error create secret")
```
Comment (alena1108) [inline pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: I think we can improve the error message a bit: `Error creating a pipeline secret`

File: pkg/pipeline/engine/jenkins/mappper.go [hunk 2]
```diff
@@ -293,25 +293,46 @@ func injectResources(container *v1.Container, cpuLimit string, cpuRequest string
 		if container.Resources.Limits == nil {
 			container.Resources.Limits = v1.ResourceList{}
 		}
-		container.Resources.Limits[v1.ResourceCPU] = resource.MustParse(cpuLimit)
+		quantity, err := resource.ParseQuantity(cpuLimit)
+		if err != nil {
+			logrus.Warnf("invalid cpuLimit %q, ignoring the config", cpuLimit)
```
Comment (alena1108) [inline pkg/pipeline/engine/jenkins/mappper.go]: do we need to return the error?

Comment (gitlawr) [reply pkg/pipeline/engine/jenkins/mappper.go]: Updated to return the error

**Comment by loganhz** [comment]
Please rebase the PR.

**Comment by gitlawr** [comment]
Rebased

**Comment by alena1108** [comment]
LGTM

**Pull request merged**